crates/sui-framework/packages/sui-framework/sources/table_vec.move

```diff
@@ -81,4 +81,14 @@ module sui::table_vec {
         table::drop(contents)
     }
 
+    /// Swaps the elements at the `i`th and `j`th indices in the TableVec `t`.
+    /// Aborts if `i` or `j` is out of bounds.
+    public fun swap<Element: store>(t: &mut TableVec<Element>, i: u64, j: u64) {
+        assert!(length(t) > i, EIndexOutOfBound);
+        assert!(length(t) > j, EIndexOutOfBound);
```

Comment:
You will also need a test that `i != j` or you will get an abort later on.
```suggestion
        assert!(length(t) > j, EIndexOutOfBound);
        if i == j { return; }
```

Reply:
Updated!

There is no `remove` function because I think it can be done with `swap' with  `i` th  and last, then `pop_back`